overlord/devicestate/handlers.go

```diff
@@ -703,3 +708,131 @@ func fetchKeys(st *state.State, keyID string) (errAcctKey error, err error) {
 		keyID = a.SignKeyID()
 	}
 }
+
+func snapState(st *state.State, name string) (*snapstate.SnapState, error) {
+	var snapst snapstate.SnapState
+	err := snapstate.Get(st, name, &snapst)
+	if err != nil && err != state.ErrNoState {
+		return nil, err
+	}
+	return &snapst, nil
+}
+
+func makeRollbackDir(name string) (string, error) {
+	rollbackDir := filepath.Join(dirs.SnapRollbackDir, name)
+
+	if err := os.MkdirAll(dirs.SnapRollbackDir, 0755); err != nil {
```

Comment:
Silly question, why not just os.MkdirAll(rollbackDir, 0755) here instead of doing it in two steps? The permissions are the same and MkdirAll ignores os.IsExists already AFAICT. Or do the two dirs need different permissions ?

Reply:
Rolling this into one. This used to be a separate if with an error.